packages/webapp/webapp_server.js

```diff
@@ -812,22 +804,82 @@ var runWebAppServer = function () {
     WebAppInternals.generateBoilerplate();
 
     // only start listening after all the startup code has run.
-    var localPort = WebAppInternals.parsePort(process.env.PORT) || 0;
+    var localPort = process.env.PORT || 0;
     var host = process.env.BIND_IP;
     var localIp = host || '0.0.0.0';
-    httpServer.listen(localPort, localIp, Meteor.bindEnvironment(function() {
-      if (process.env.METEOR_PRINT_ON_LISTEN)
-        console.log("LISTENING"); // must match run-app.js
 
-      var callbacks = onListeningCallbacks;
-      onListeningCallbacks = null;
-      _.each(callbacks, function (x) { x(); });
+    var startHttpServer = function(listenOptions) {
+      httpServer.listen(listenOptions, Meteor.bindEnvironment(function() {
+        if (process.env.METEOR_PRINT_ON_LISTEN)
+          console.log("LISTENING"); // must match run-app.js
+
+        var callbacks = onListeningCallbacks;
+        onListeningCallbacks = null;
+        _.each(callbacks, function (x) { x(); });
+
+      }, function (e) {
+        console.error("Error listening:", e);
+        console.error(e && e.stack);
+      }));
+    };
+
+    localPort = isNaN(Number(localPort)) ? localPort : Number(localPort);
 
-    }, function (e) {
-      console.error("Error listening:", e);
-      console.error(e && e.stack);
-    }));
+    if (typeof localPort == "string") {
+      var socketPath = localPort;
+      var listenOptions = { path: socketPath };
 
+      // Test to see if the socket path is Windows Server-style named pipe with the
+      // format: \\.\pipe\{randomstring} or \\{servername}\pipe\{randomstring}
+      // Otherwise, treat it as a UNIX domain socket path pointing to a socket file
+      if ( /\\\\?.+\\pipe\\?.+/.test(socketPath)) {
+        startHttpServer(listenOptions);
+      } else {
+        if (fs.existsSync(socketPath)) {
+          if (fs.statSync(socketPath).isSocket()) {
+            var clientSocket = new net.Socket();
+            clientSocket.on('error', Meteor.bindEnvironment(function(e) {
+              //If there is an existing socket file that we cannot connect to,
+              //it must be stale as it is not associated with a running server
+              if (e.code == 'ECONNREFUSED') {
+                console.log("Deleting stale socket file");
+                fs.unlinkSync(socketPath);
+                startHttpServer(listenOptions);
+              }
+            }));
+            clientSocket.connect({ path: socketPath }, function() {
```

Comment:
net.createConnection() is not suitable here because it is necessary to set the error handler in advance for this very short-lived client socket object whose sole purpose is to distinguish between a live UNIX socket and a stale UNIX socket file.

The documentation does not discourage it use, but says "ise this only when implementing a custom Socket." or in my view "use it when you have to".